### ISSUE TITLE
Let's discuss a text alternative for the loading spinner

### ISSUE THREAD

**Issue by misteroneill** [videojs/video.js#4902]
I'm particularly interested in perspective from @OwenEdwards.

Currently, there is no text alternative for the loading spinner to reflect that a `waiting` event has occurred. It seems like this could be a useful accessibility measure, but the precise implementation of it is an open question. Here are some ideas to get discussion going:

1. Add a visually-hidden `role="alert"` element to the player that should announce that the player is loading/buffering.
1. Add a visually-hidden status element immediately before the control bar that contains the current status of the player if it's interesting.
1. Any other ideas?

The loading spinner shows on seeking, but I don't think we should interrupt seeking when buffering is expected in that case. At least, not until seeking is finished and the player is buffering at the seeked-to point.

**Comment by Jad3z** [comment]
+1

**Comment by gkatsev** [comment]
I think the discussion here is a bit larger than the loading spinner specifically but rather a more general "how do we notify the user, particularly, one that is using a screen reader, of state changes in the player"? For sighted users, a loading spinner is generally a good indicator that something is happening wrt loading of the video but currently our loading spinner has no text alternative for screen readers. Adding a text alternative to it, I think, is the first step to this and should be done regardless of how we choose to address the notification issue.

**Comment by gkatsev** [comment]
Just merged #4916 which adds a text alternative to the loading spinner. This doesn't fully solve the issues here but gets us one step closer to making things better.

**Comment by gkatsev** [comment]
So, circling back to this, I don't think we want or need an alternative. Instead, we want to have the loading spinner announce the state to the screen reader, if the user is focused inside the player. Then, if the spinner is continuously up for more than a certain time, it should announce again, for example, every 10 seconds.

We don't have many other things that we need to provide status for to screen readers that don't already do so themselves.

**Comment by OwenEdwards** [comment]
This example is probably the way to go:

http://pauljadam.com/demos/loading-indicator.html

**Comment by OwenEdwards** [comment]
Here's another reference on this:

https://codeburst.io/how-to-create-a-simple-css-loading-spinner-make-it-accessible-e5c83c2e464c

Although we should note that in some cases it makes sense to announce to a screen reader user that something in the page is loading, but it *doesn't* make sense if they don't have the video player (or part of it) focused *and* they are interacting with or reading another part of the page.
